src/amo/components/UserProfile/index.js

```diff
@@ -141,9 +142,9 @@ export class UserProfileBase extends React.Component<InternalProps> {
   }
 
   getURL() {
-    const { params } = this.props;
+    const { params, user } = this.props;
 
-    return `/user/${params.username}/`;
+    return `/user/${user ? user.username : params.username}/`;
```

Comment:
Because you have this logic three times in the file, might it be worth creating a `getUsername` helper that returns `user ? user.username : params.username`?

Also, I'm not sure if it matters (I imagine not), but this will also cause the links in the `Paginate` component and the `AddonsByAuthorsCard`s to be converted from numeric ids into usernames. I.e., if I visit `/user/12345/`, the links will be to `/user/some-username/`.

Reply:
Yes, I think that's fine if we link to username in the `Paginate` component. The main idea is to allow the legacy URLs to be served.